### PR TITLE
Update to latest sdk version and priovide new api of CalculatePrice and Purchase

### PR DESCRIPTION
**Pull request by Only2125**
## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

**Comment by Only2125** [comment]
Hi is there a service error? I just go over those errors, it seems not from my side.

**Comment by isra-fel** [comment]
The ci fails at the step of generating help. Can you generate locally? @Only2125

**Comment by Only2125** [comment]
> The ci fails at the step of generating help. Can you generate locally? @Only2125

Hi @isra-fel , it shows the same error. How can I fix it?

**Comment by msJinLei** [comment]
> > The ci fails at the step of generating help. Can you generate locally? @Only2125
> 
> Hi @isra-fel , it shows the same error. How can I fix it?

@Only2125 Please see that error message and refer to the webpage https://docs.microsoft.com/en-us/powershell/scripting/developer/cmdlet/approved-verbs-for-windows-powershell-commands?view=powershell-6

````Unhandled Exception: System.InvalidOperationException: Could not parse '"D:\a\1\s\artifacts\Debug\Az.Reservations\Microsoft.Azure.PowerShell.Cmdlets.Reservations.dll,","Microsoft.Azure.Commands.Reservations.Cmdlets.Calculate,","Calculate-AzCalculatePrice, ","1,","8300,","Calculate-AzCalculatePrice uses the verb 'Calculate', which is not on the list of approved verbs for PowerShell commands. Use the cmdlet 'Get-Verb' to see the full list of approved verbs and consider renaming the cmdlet.,",Consider renaming the cmdlet to use an approved verb for PowerShell.' as HelpIssue record````

**Comment by msJinLei** [comment]
@Only2125 Please Since you have added a new cmdlet. Please submit a design review firstly.

**Comment by msJinLei** [comment]
@Only2125 Since you have added a new cmdlet. Please submit a [design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)

**Comment by Only2125** [comment]
https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/436
Send a request for the Cmdlet Review

File: tools/StaticAnalysis/Exceptions/Az.Reservations/SignatureIssues.csv [hunk 1]
```diff
@@ -1 +1 @@
-"AssemblyFileName","ClassName","Target","Severity","ProblemId","Description","Remediation"
```
Comment (msJinLei) [inline tools/StaticAnalysis/Exceptions/Az.Reservations/SignatureIssues.csv]: Remove unnecessary change.
